torch/csrc/jit/runtime/graph_executor.cpp

```diff
@@ -919,7 +920,10 @@ void runOptimization(
       "After EliminateDeadCode, before EliminateCommonSubexpression\n", *graph);
   EliminateCommonSubexpression(graph);
   GRAPH_DEBUG(
-      "After EliminateCommonSubexpression, before PeepholeOptimize\n", *graph);
+      "After EliminateCommonSubexpression, before HoistCommonExpression\n",
+      *graph);
+  HoistCommonExpression(graph);
```

Comment:
i think just running this once in the pass is fine. this is a very targeted optimization to try to handle the case of 
```
if inplace:
    y = x.add_(3)
else:
    y = x.add(3)
```

which we want to rewrite to 
```
if inplace:
    y = x.add(3)
else:
    y = x.add(3)
```
and hoist out. we're not running the mutation removal yet, but we should be shortly... cc @desertfire make sure these line up with the mutation removal pass. 

we're not really expecting any juice out of this pass other than the inplace handling so just once fine probably